core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -714,12 +731,26 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
     }
 
     /** Everything that happens to both sides equally when war is declared by one side on the other */
-    private fun onWarDeclared() {
+    private fun onWarDeclared(isOffensiveWar: Boolean) {
         // Cancel all trades.
         for (trade in trades)
             for (offer in trade.theirOffers.filter { it.duration > 0 })
                 civInfo.addNotification("[${offer.name}] from [$otherCivName] has ended",
                     NotificationCategory.Trade, otherCivName, NotificationIcon.Trade)
+
+        // If we attacked, then we need to end all of our defensive pacts acording to Civ 5
+        if (isOffensiveWar) {
+            civInfo.diplomacy.values.filter { it.otherCiv() != otherCiv() // So that we don't cancel the defensive pact before negative modifiers are added
+                && it.diplomaticStatus == DiplomaticStatus.DefensivePact }.forEach {
+                // Trades with defensive pact are now invalid
+                val defensivePactOffer = it.trades.firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact} }
+                it.trades.remove(defensivePactOffer)
```

Comment:
Needs to be cancelled for both sides. See example of removeUntenebleTrades.
Also we need notifications for both sides